tardis/plasma/tests/test_nlte_solver.py

```diff
@@ -71,3 +162,55 @@ def test_rate_matrix():
     assert_almost_equal(
         desired_rate_matrix, np.array(actual_rate_matrix), decimal=6
     )
+
+
+def test_jacobian_matrix(
+    simple_phi,
+    simple_electron_density,
+    simple_rate_matrix_index,
+    simple_total_photo_ion_coefficients,
+    simple_total_rad_recomb_coefficients,
+    simple_total_col_ion_coefficients,
+    simple_total_col_recomb_coefficients,
+):
+    """
+    Using a simple case of nlte_ion for HI and HeII,
+    checks if the jacobian_matrix generates the correct data.
+    """
+
+    initial_guess = [
+        0.7192433675307516,
+        0.8101666197902874,
+        0.7171853313284426,
+        0.040220760173800496,
+        0.2878574499274399,
+        simple_electron_density,
+    ]
+    simple_rate_matrix = NLTERateEquationSolver.calculate_rate_matrix(
+        simple_phi,
+        simple_electron_density,
+        simple_rate_matrix_index,
+        simple_total_photo_ion_coefficients,
+        simple_total_rad_recomb_coefficients,
+        simple_total_col_ion_coefficients,
+        simple_total_col_recomb_coefficients,
+    )
+
+    actual_jacobian_matrix = NLTERateEquationSolver.jacobian_matrix(
+        initial_guess,
+        simple_rate_matrix,
+        simple_rate_matrix_index,
+        simple_total_rad_recomb_coefficients,
+        simple_total_col_ion_coefficients,
+        simple_total_col_recomb_coefficients,
+    )
+
+    desired_jacobian_matrix = [
+        [-0.07760098, 0.09927163, 0.0, 0.0, 0.0, 0.0],
```

Comment:
Why is the last element zero? H I is treated with NLTE ionization? Have you tuned the inputs to achieve this? I am too lazy to check the math myself ;-)